### PR TITLE
Relax 'Maximum' buffer tier to 7GB RAM

### PR DESCRIPTION
**Pull request by marcelveldt**
# What does this implement/fix?

Follow-up to #4249. The **Maximum** buffer preset (and the auto-selected default) required exactly **8GB** of RAM. Machines marketed as "8GB" frequently report a little less — integrated GPUs carve out shared system memory (e.g. a real 7.7GB), and container runtimes report their cgroup limit — so they silently dropped to the Balanced tier and never got the largest buffer.

- Lower the **Maximum** threshold from 8GB to **7GB**, so near-8GB hosts qualify for the optimal buffer.
- Keep the **4GB Balanced** minimum intact (the genuine lower bound).
- Replace the repeated magic numbers with named `BALANCED_MIN_RAM_GB` / `MAXIMUM_MIN_RAM_GB` constants and extend the buffer-tier test with the boundary cases (6.9 → Balanced, 7.0 / 7.7 → Maximum).

**Related issue (if applicable):**

- n/a

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue) — `bugfix`
- [ ] New feature (non-breaking change which adds functionality) — `new-feature`
- [x] Enhancement to an existing feature — `enhancement`
- [ ] New music/player/metadata/plugin provider — `new-provider`
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) — `breaking-change`
- [ ] Refactor (no behaviour change) — `refactor`
- [ ] Documentation only — `documentation`
- [ ] Maintenance / chore — `maintenance`
- [ ] CI / workflow change — `ci`
- [ ] Dependencies bump — `dependencies`

## Checklist

- [x] The code change is tested and works locally.
- [x] `pre-commit run --all-files` passes.
- [x] `pytest` passes, and tests have been added/updated under `tests/` where applicable.
- [ ] For changes to shared models, the companion PR in `music-assistant/models` is linked.
- [ ] For changes affecting the UI, the companion PR in `music-assistant/frontend` is linked.
- [x] I have read and complied with the project's [AI Policy](https://github.com/music-assistant/.github/blob/main/AI_POLICY.md) for any AI-assisted contributions.
- [ ] I have raised a PR against the documentation repository targeting the main or beta branch as appropriate.

### CODE REVIEW

File: tests/controllers/streams/test_buffer_size.py [hunk 1]
```diff
@@ -15,13 +15,18 @@
...
         (16.0, [BufferSize.MINIMAL, BufferSize.BALANCED, BufferSize.MAXIMUM]),
         # unknown memory (0.0) -> offer everything (fail open)
         (0.0, [BufferSize.MINIMAL, BufferSize.BALANCED, BufferSize.MAXIMUM]),
     ],
 )
 def test_get_available_buffer_sizes(total_gb: float, expected: list[BufferSize]) -> None:
-    """Balanced requires >= 4GB, Maximum >= 8GB; unknown RAM offers all (fail open)."""
+    """Balanced requires >= 4GB, Maximum >= 7GB; unknown RAM offers all (fail open)."""
     with patch.object(constants, "TOTAL_SYSTEM_MEMORY_GB", total_gb):
         assert constants.get_available_buffer_sizes() == expected
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply tests/controllers/streams/test_buffer_size.py:32]: Good catch, added a test for the default selection too.
